packages/figma-utils/src/utils/generate.ts

```diff
@@ -1,40 +1,25 @@
 import { ParsedVariable } from "../types/figma.js";
 
-type GenericGeneratorOptions = {
-  data: ParsedVariable;
-  /**
-   * Function which returns the full variable name depending on the used format.
-   *
-   * @example
-   * ```ts
-   * // for CSS:
-   * (name) => `--${name}`
-   * ```
-   */
-  nameTransformer: (name: string) => string;
-  /**
-   * Function which returns the full variable value if its an alias / reference to another variable.
-   *
-   * @example
-   * ```ts
-   * // for CSS:
-   * (name) => `var(--${name})`
-   * ```
-   */
-  aliasTransformer: (name: string) => string;
-};
-
 /**
  * Generates the given parsed Figma variables into CSS variables.
  *
+ * @param data Parsed Figma variables
+ * @param selector CSS selector to use for the CSS format. The mode name will be added to the selector
+ * if it is set to something other than ":root", e.g. for the mode named "dark", passing the selector "html" will result in "html.dark"
  * @returns File content of the .css file
  */
-export const generateAsCSS = (data: ParsedVariable): string => {
-  return genericGenerator({
-    data,
-    nameTransformer: (name) => `--${name}`,
-    aliasTransformer: (name) => `var(--${name})`,
+export const generateAsCSS = (data: ParsedVariable, selector: string = ":root"): string => {
+  const variableContent = Object.entries(data.variables).map(([name, value]) => {
+    const { isAlias, variableName } = isAliasVariable(value);
+    const variableValue = isAlias ? `var(--${variableName})` : value;
+    return `  --${name}: ${variableValue};`;
   });
+
+  selector = selector.trim();
+  if (selector !== ":root") selector += `.${data.modeName}`;
+
+  return `${generateTimestampComment(data.modeName)}
+${selector} {\n${variableContent.join("\n")}\n}\n`;
```

Comment:
isn't it advised against to change func params directly?
```suggestion
  let fullSelector = selector.trim();
  if (fullSelector !== ":root") fullSelector += `.${data.modeName}`;

  return `${generateTimestampComment(data.modeName)}
${fullSelector} {\n${variableContent.join("\n")}\n}\n`;
```

Reply:
Done